File: php/features.php
<?php
declare(strict_types=1);

// DomainWhois SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class DomainWhoisFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new DomainWhoisBaseFeature();
            case "test":
                return new DomainWhoisTestFeature();
            default:
                return new DomainWhoisBaseFeature();
        }
    }
}

File: py/domainwhois_sdk/features.py
# DomainWhois SDK feature factory

from domainwhois_sdk.feature.base_feature import DomainWhoisBaseFeature
from domainwhois_sdk.feature.test_feature import DomainWhoisTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DomainWhoisBaseFeature(),
        "test": lambda: DomainWhoisTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
